### PR TITLE
Avoid using RPM/DEB based installation and install managed plugin

### LINKED ISSUE #1051: Complete the move to managed plugins - drop support for unmanaged ones
In #1006 we added managed plugins support
To complete this, we already have #1033 (photoOS support) and #1020 (CI / full build) issues. 

We also need to 

- [ ] drop build and use of .rpm/.deb. We can still have the targets in Makefile in case we need it for troubleshooting or comparison, but they should not be a part of regular build. 
- [ ] change all tests to use the managed plugins (since in #1020 we focus only ADDING management plugin test) 
- [x] update documentation to refer to Docker managed plugin in, and give an example of vDVS install, plus debug option (fixed #1115)

Few more action items (notes from @shuklanirdesh82)

- [x] upgrade all test VMs to use docker 1.13/17.03 (keep last two docker releases)
- [x] install ```make``` utility to all VMs

//CC @shuklanirdesh82 @tusharnt

### PR DESCRIPTION
**Pull request by shuklanirdesh82**
fixes #1051: Complete the move to managed plugins - drop support for unmanaged ones

This PR addresses following things

1. `make build-all` builds managed plugin as well as push to docker hub after building
2. `deploy-vm` helper is corrected
3. new helper method `installManagedPlugin` for installing plugin
4. `cleanVM` helper is updated to remove managed plugin if installed
5. corrected tests to accommodate such change

Tested locally: (ran ``` make e2e-test```)

```
nshukla-mbp:docker-volume-vsphere nshukla$ make clean-all build-all deploy-all
Makefile:54: warning: overriding commands for target `clean'
plugin/Makefile:88: warning: ignoring old commands for target `clean'
/Library/Developer/CommandLineTools/usr/bin/make --directory=vmdk_plugin clean-all
../misc/scripts/build.sh clean-as-root
make: Entering directory `/go/src/github.com/vmware/docker-volume-vsphere/vmdk_plugin'
rm -rf ../build .code_verify
make --directory=../esx_service clean
make[1]: Entering directory `/go/src/github.com/vmware/docker-volume-vsphere/esx_service'
rm -rf vmware-esx-vmdkops-0.13.90eea6a.vib ../build/libvmci_srv.so ../build/libvmci_srv.so.64 ../build/descriptor.xml ../build/payloads .lint_*
make[1]: Leaving directory `/go/src/github.com/vmware/docker-volume-vsphere/esx_service'
make: Leaving directory `/go/src/github.com/vmware/docker-volume-vsphere/vmdk_plugin'
../misc/scripts/deploy-tools.sh cleanvm "10.20.104.62 10.20.104.62" ""vsphere:latest""


=> cleanvm: Cleaning up on root@10.20.104.62 : deb Wed Apr 12 17:14:47 PDT 2017


dpkg-query: no packages found matching docker-volume-vsphere
vsphere:latest
../misc/scripts/deploy-tools.sh cleanesx "10.20.105.97" esx-vmdkops-service


=> Cleaning up on ESX 10.20.105.97 Wed Apr 12 17:14:48 PDT 2017


Removal Result:
   Message: Operation finished successfully.
   Reboot Required: false
   VIBs Installed: 
   VIBs Removed: VMWare_bootbank_esx-vmdkops-service_0.13.90eea6a-0.0.1
   VIBs Skipped: 
/Library/Developer/CommandLineTools/usr/bin/make --directory=vmdk_plugin build-all
../misc/scripts/build.sh pylint
make: Entering directory '/go/src/github.com/vmware/docker-volume-vsphere/esx_service'
pylint --errors-only -d import-error,no-name-in-module --rcfile pylint_cfg.ini ../esx_service
pylint --errors-only -d import-error,no-name-in-module --rcfile pylint_cfg.ini ../esx_service/vmodl
pylint --errors-only -d import-error,no-name-in-module --rcfile pylint_cfg.ini cli
pylint --errors-only -d import-error,no-name-in-module --rcfile pylint_cfg.ini utils
make: Leaving directory '/go/src/github.com/vmware/docker-volume-vsphere/esx_service'
../misc/scripts/build.sh build
make: Entering directory `/go/src/github.com/vmware/docker-volume-vsphere/vmdk_plugin'
go tool vet .
gofmt -s -l -w .
GO15VENDOREXPERIMENT=1 go build --ldflags '-extldflags "-static"' -o ../build/docker-volume-vsphere github.com/vmware/docker-volume-vsphere/vmdk_plugin
GO15VENDOREXPERIMENT=1 go test -c -o ../build/vmdkops.test github.com/vmware/docker-volume-vsphere/vmdk_plugin/drivers/vmdk/vmdkops -cover
GO15VENDOREXPERIMENT=1 go test -c -o ../build/docker-volume-vsphere.test github.com/vmware/docker-volume-vsphere/vmdk_plugin -cover
make  --directory=../esx_service build
make[1]: Entering directory `/go/src/github.com/vmware/docker-volume-vsphere/esx_service'
cc -fPIC -m32 -shared -o ../build/libvmci_srv.so vmci/vmci_server.c
cc -fPIC -shared -o ../build/libvmci_srv.so.64 vmci/vmci_server.c
Staging in ../build/payloads
cp __init__.py vmdk_ops.py volume_kv.py vsan_policy.py cli/vmdkops_admin.py cli/vmdkops_post_update.11.1.py ../build/payloads/vmdkops/usr/lib/vmware/vmdkops/bin
cp utils/__init__.py utils/auth.py utils/auth_api.py utils/auth_data.py utils/auth_data_const.py utils/cli_table.py utils/convert.py utils/error_code.py utils/kvESX.py utils/log_config.py utils/threadutils.py utils/vmdk_utils.py utils/vsan_info.py  ../build/payloads/vmdkops/usr/lib/vmware/vmdkops/Python
cp -R tools/sqlite/sqlite3 ../build/payloads/vmdkops/usr/lib/vmware/vmdkops/Python/2
cp tools/sqlite/*.so ../build/payloads/vmdkops/usr/lib/vmware/vmdkops/Python/2
cp ../build/libvmci_srv.so ../build/payloads/vmdkops/usr/lib/vmware/vmdkops/lib/libvmci_srv.so
cp ../build/libvmci_srv.so.64 ../build/payloads/vmdkops/usr/lib/vmware/vmdkops/lib64/libvmci_srv.so
cp vmdk-opsd ../build/payloads/vmdkops/etc/init.d
VMDKOPS_PAYLOAD=../build/payloads/vmdkops make  --directory=vmodl copy_to_payload
make[2]: Entering directory `/go/src/github.com/vmware/docker-volume-vsphere/esx_service/vmodl'
Ignoring VMODL 'copy_to_payload' target.
See https://github.com/vmware/docker-volume-vsphere/pull/975 for details.
make[2]: Leaving directory `/go/src/github.com/vmware/docker-volume-vsphere/esx_service/vmodl'
vibauthor --debug --compose --vib=../build/vmware-esx-vmdkops-0.13.90eea6a.vib --stage-dir ../build --offline-depot ../build/vmware-esx-vmdkops-0.13.90eea6a.zip --force
WARNING: a VIB file with "accepted" acceptance level need to be signed
    you can do it later using vibauthor with --sign option
Successfully created ../build/vmware-esx-vmdkops-0.13.90eea6a.vib.
Successfully created ../build/vmware-esx-vmdkops-0.13.90eea6a.zip.
make[1]: Leaving directory `/go/src/github.com/vmware/docker-volume-vsphere/esx_service'
make: Leaving directory `/go/src/github.com/vmware/docker-volume-vsphere/vmdk_plugin'
../misc/scripts/build.sh deb
make: Entering directory '/go/src/github.com/vmware/docker-volume-vsphere/vmdk_plugin'
fpm --deb-no-default-config-files -p ../build -C package -s dir -n docker-volume-vsphere -v 0.13.90eea6a --provides docker-volume-vsphere -m cna-storage@vmware.com --url https://github.com/vmware/docker-volume-vsphere --after-install ../misc/scripts/install/after-install.sh --before-remove ../misc/scripts/install/before-remove.sh --after-remove ../misc/scripts/install/after-remove.sh --description "Docker Volume Driver for vSphere" --architecture x86_64 --force -t deb .
Created package {:path=>"../build/docker-volume-vsphere_0.13.90eea6a_amd64.deb"}
make: Leaving directory '/go/src/github.com/vmware/docker-volume-vsphere/vmdk_plugin'
../misc/scripts/build.sh rpm
make: Entering directory '/go/src/github.com/vmware/docker-volume-vsphere/vmdk_plugin'
fpm --epoch 0 -p ../build -C package -s dir -n docker-volume-vsphere -v 0.13.90eea6a --provides docker-volume-vsphere -m cna-storage@vmware.com --url https://github.com/vmware/docker-volume-vsphere --after-install ../misc/scripts/install/after-install.sh --before-remove ../misc/scripts/install/before-remove.sh --after-remove ../misc/scripts/install/after-remove.sh --description "Docker Volume Driver for vSphere" --architecture x86_64 --force -t rpm .
Created package {:path=>"../build/docker-volume-vsphere-0.13.90eea6a-1.x86_64.rpm"}
make: Leaving directory '/go/src/github.com/vmware/docker-volume-vsphere/vmdk_plugin'
/Library/Developer/CommandLineTools/usr/bin/make --directory=plugin all
Using the following config:
DOCKER_HUB_REPO shuklanirdesh EXTRA_TAG -dev VERSION_TAG 0.14
PLUGIN_NAME shuklanirdesh/docker-volume-vsphere:0.14-dev
=== Cleaning work files, images and plugin(s)...
rm -rf ./rootfs
rm -f docker-volume-vsphere
docker plugin rm shuklanirdesh/docker-volume-vsphere:0.14-dev -f
shuklanirdesh/docker-volume-vsphere:0.14-dev
docker rmi shuklanirdesh/docker-volume-vsphere:rootfs
Error response from daemon: No such image: shuklanirdesh/docker-volume-vsphere:rootfs
make[1]: [clean] Error 1 (ignored)
== building Docker image, unpacking to ./rootfs and creating plugin...
cp ../build/docker-volume-vsphere .
docker build -q -t shuklanirdesh/docker-volume-vsphere:rootfs .
sha256:9ca0f6c4ea031dc8d0a03d31ebd9013052286e64203787000f393d2c14f210e0
mkdir -p ./rootfs
docker create --name tempContainer shuklanirdesh/docker-volume-vsphere:rootfs
29d32963e25aa84faf1edd8ddc3cbb6aed26bf6d69e7aa438ad2e5d8fc89d72c
docker export  tempContainer  | tar -x -C ./rootfs
docker rm -vf  tempContainer
tempContainer
docker rmi shuklanirdesh/docker-volume-vsphere:rootfs
Untagged: shuklanirdesh/docker-volume-vsphere:rootfs
Deleted: sha256:9ca0f6c4ea031dc8d0a03d31ebd9013052286e64203787000f393d2c14f210e0
Deleted: sha256:b2052c66ddf53ebe3d1ffdfbb03aaacf728d93795e92be1eb72506feaaa83da7
Deleted: sha256:5f76289b756457a2ccb861a5477dcc3b244b635359388b2ab9ddb47041458abc
Deleted: sha256:8e23b23254748b2e14e76339c43589c6f4ca057c5dd0ad5ee272f531c194f402
Deleted: sha256:c0855efdb930680f266ce0b6552a5a1092fd9ee87ff3ec9f0a70ad80c04a2747
Deleted: sha256:a9dd24688f71cd098fb800fbf197b8d1945e98aa49e4c4b5c44e80de5ecfa5b7
Deleted: sha256:5d78ac447d700d797470105a5e3bf8e83672d7e0bbfc05663c43eef41facb6a7
-- Creating  plugin shuklanirdesh/docker-volume-vsphere:0.14-dev ...
docker plugin create shuklanirdesh/docker-volume-vsphere:0.14-dev .
shuklanirdesh/docker-volume-vsphere:0.14-dev
Pushing shuklanirdesh/docker-volume-vsphere:0.14-dev to dockerhub.io...
The push refers to a repository [docker.io/shuklanirdesh/docker-volume-vsphere]
c2f766e6d3bd: Pushed 
0.14-dev: digest: sha256:9d61c99d78d0ecd622e221cf37822aff975ac4fad0e5de440b9b2a5ff10bdc77 size: 519
/Library/Developer/CommandLineTools/usr/bin/make --directory=vmdk_plugin deploy-all
../misc/scripts/deploy-tools.sh cleanesx "10.20.105.97" esx-vmdkops-service


=> Cleaning up on ESX 10.20.105.97 Wed Apr 12 17:17:19 PDT 2017


../misc/scripts/deploy-tools.sh deployesx "10.20.105.97" "../build/vmware-esx-vmdkops-0.13.90eea6a.vib"


=> Deploying to ESX root@10.20.105.97 Wed Apr 12 17:17:22 PDT 2017


Installation Result:
   Message: Operation finished successfully.
   Reboot Required: false
   VIBs Installed: VMWare_bootbank_esx-vmdkops-service_0.13.90eea6a-0.0.1
   VIBs Removed: 
   VIBs Skipped: 
vmdkops-opsd is running pid=1083647
../misc/scripts/deploy-tools.sh cleanvm "10.20.104.62 10.20.104.62" ""vsphere:latest""


=> cleanvm: Cleaning up on root@10.20.104.62 : deb Wed Apr 12 17:17:51 PDT 2017


dpkg-query: no packages found matching docker-volume-vsphere
Error response from daemon: plugin "vsphere:latest" not found
../misc/scripts/deploy-tools.sh deployvm "10.20.104.62 10.20.104.62" shuklanirdesh/docker-volume-vsphere:0.14-dev


=> Deploying root@10.20.104.62 : deb Wed Apr 12 17:17:53 PDT 2017




=> installManagedPlugin: Installing vDVS plugin [shuklanirdesh/docker-volume-vsphere:0.14-dev] Wed Apr 12 17:17:53 PDT 2017


0.14-dev: Pulling from shuklanirdesh/docker-volume-vsphere
8bf57e2daba8: Verifying Checksum
8bf57e2daba8: Download complete
Digest: sha256:9d61c99d78d0ecd622e221cf37822aff975ac4fad0e5de440b9b2a5ff10bdc77
Status: Downloaded newer image for shuklanirdesh/docker-volume-vsphere:0.14-dev
Installed plugin shuklanirdesh/docker-volume-vsphere:0.14-dev
28049
../misc/scripts/deploy-tools.sh deployvmtest "10.20.104.62 10.20.104.62" ../build ../misc/scripts


=> Deploying test code to root@10.20.104.62 Wed Apr 12 17:18:00 PDT 2017

  
nshukla-mbp:docker-volume-vsphere nshukla$ make e2e-test
Makefile:54: warning: overriding commands for target `clean'
plugin/Makefile:88: warning: ignoring old commands for target `clean'
/Library/Developer/CommandLineTools/usr/bin/make --directory=vmdk_plugin e2e-test

=> Running target e2e-test Wed Apr 12 17:18:57 PDT 2017

../misc/scripts/build.sh e2e-dkrVolDriver-test "10.20.105.97" "10.20.104.62" "10.20.104.62"
make: Entering directory `/go/src/github.com/vmware/docker-volume-vsphere/vmdk_plugin'

=> Running target e2e-dkrVolDriver-test Thu Apr 13 00:18:58 UTC 2017

GO15VENDOREXPERIMENT=1 go test github.com/vmware/docker-volume-vsphere/vmdk_plugin/E2E_Tests -cover -v
=== RUN   TestVolumeCreationFirstTime

creating volume [abc_1492042738] on VM[10.20.104.62]
destroying volume [abc_1492042738]
creating volume [abc_1492042745] on VM[10.20.104.62]
destroying volume [abc_1492042745]--- PASS: TestVolumeCreationFirstTime (14.20s)
	createVolume_test.go:63: 
		Testcase passed: successfully able to create volume [abc_1492042738
		]
	createVolume_test.go:63: 
		Testcase passed: successfully able to create volume [abc_1492042745
		]
=== RUN   TestVolumeProperties
2017/04/13 00:19:12 running end_to_end tests - current time:  2017-04-13 00:19:12.877165984 +0000 UTC
2017/04/13 00:19:12 Running test on VM -  10.20.104.62
2017/04/13 00:19:13 Docker version:   Docker version 17.03.0-ce, build 60ccb22

2017/04/13 00:19:13 Creating a volume of Format Type -  thin
2017/04/13 00:19:18 
 successfully created a  volume --  dockerVol_1492042753
2017/04/13 00:19:18 Verifying volume properties like size, disk-format and attached-to-vm fields at vm and esx for volume -  dockerVol_1492042753
2017/04/13 00:19:27 Finished verifying volume properties like size, disk-format and attached-to-vm fields at vm and esx for volume -  dockerVol_1492042753
2017/04/13 00:19:38 Creating a volume of Format Type -  zeroedthick
2017/04/13 00:19:44 
 successfully created a  volume --  dockerVol_1492042778
2017/04/13 00:19:44 Verifying volume properties like size, disk-format and attached-to-vm fields at vm and esx for volume -  dockerVol_1492042778
2017/04/13 00:19:53 Finished verifying volume properties like size, disk-format and attached-to-vm fields at vm and esx for volume -  dockerVol_1492042778
2017/04/13 00:20:04 Creating a volume of Format Type -  eagerzeroedthick
2017/04/13 00:20:11 
 successfully created a  volume --  dockerVol_1492042804
2017/04/13 00:20:11 Verifying volume properties like size, disk-format and attached-to-vm fields at vm and esx for volume -  dockerVol_1492042804
2017/04/13 00:20:20 Finished verifying volume properties like size, disk-format and attached-to-vm fields at vm and esx for volume -  dockerVol_1492042804
--- PASS: TestVolumeProperties (78.52s)
PASS
coverage: 0.0% of statements
2017/04/13 00:20:34 -----Clean-up finished - current time:  2017-04-13 00:20:34.207652826 +0000 UTC
ok  	github.com/vmware/docker-volume-vsphere/vmdk_plugin/E2E_Tests	95.530s
make: Leaving directory `/go/src/github.com/vmware/docker-volume-vsphere/vmdk_plugin'
```

### CODE REVIEW

File: vmdk_plugin/E2E_Tests/e2e_verifyVolmProperties_test.go [hunk 1]
```diff
@@ -76,7 +76,7 @@ func TestVolumeProperties(t *testing.T) {
 				containerName = "busybox_" + strconv.FormatInt(time.Now().Unix(), 20)
 				log.Println("Creating a volume of Format Type - ", formatTypes[k])
 				volName := TestInputParamsUtil.GetVolumeNameWithTimeStamp("dockerVol")
-				_, err := TestUtil.InvokeCommand(vms[vmIndx], "docker volume create --driver=vsphere --name="+
+				_, err := TestUtil.InvokeCommand(vms[vmIndx], "docker volume create --driver=vsphere:latest --name="+
```
Comment (msterin) [inline vmdk_plugin/E2E_Tests/e2e_verifyVolmProperties_test.go]: we should not need :latest, it is picked up by default

File: vmdk_plugin/sanity_test.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ func init() {
 	configFile := flag.String("config", config.DefaultConfigPath, "Configuration file path")
 
 	flag.BoolVar(&removeContainers, "rm", true, "rm container after run")
-	flag.StringVar(&driverName, "d", "vsphere", "Driver name. We refcount volumes on this driver")
+	flag.StringVar(&driverName, "d", "vsphere:latest", "Driver name. We refcount volumes on this driver")
```
Comment (msterin) [inline vmdk_plugin/sanity_test.go]: it should accept ismply "vsphere" so no changes should be here

File: vmdk_plugin/utils/test_util/VolumeCreationTestUtil.go [hunk 3]
```diff
@@ -30,7 +30,7 @@ var SSH_OPTS []string = []string{strings.Split(os.Getenv("SSH_KEY_OPT"), " ")[0]
 // defaults.
 func CreateDefaultVolume(ip string, name string) ([]byte, error) {
 	fmt.Printf("\ncreating volume [%s] on VM[%s]", name, ip)
-	return InvokeCommand(ip, "docker volume create --driver=vsphere --name="+name)
+	return InvokeCommand(ip, "docker volume create --driver=vsphere:latest --name="+name)
 }
```
Comment (msterin) [inline vmdk_plugin/utils/test_util/VolumeCreationTestUtil.go]: same comment, there should be no changes in this line

File: vmdk_plugin/E2E_Tests/e2e_verifyVolmProperties_test.go (hunk 1 above)
Comment (shuklanirdesh82) [inline vmdk_plugin/E2E_Tests/e2e_verifyVolmProperties_test.go]: it complains when `:latest` not passed.

**Comment by shuklanirdesh82** [comment]
Thanks @msterin  for the quick review; please find my comment inline.

> Why are we cleanup up both, is is for making transition easy ? 

Yeah, it was for making transition easy. 

> I think we can do manual step once to remove it.

I will update contributing.md (for one time removal manually) and remove the step from cleanVM part.

TODO: CONTRIBUTING.md will be added in the next diff.

File: vmdk_plugin/utils/test_util/VolumeCreationTestUtil.go (hunk 3 above)
Comment (shuklanirdesh82) [inline vmdk_plugin/utils/test_util/VolumeCreationTestUtil.go]: volume creation command fails saying not able to find "vsphere" legacy driver

Comment (msterin) [inline vmdk_plugin/utils/test_util/VolumeCreationTestUtil.go]:
```
root@ubuntu-test:~# docker plugin ls
ID                  NAME                DESCRIPTION         ENABLED
root@ubuntu-test:~# docker plugin install --alias=vsphere msterin/docker-volume-vsphere:0.14-dev
Plugin "msterin/docker-volume-vsphere:0.14-dev" is requesting the following privileges:
 - network: [none]
 - mount: [/dev]
 - mount: [/var/run]
 - mount: [/etc]
 - mount: [/var/log]
 - allow-all-devices: [true]
 - capabilities: [CAP_SYS_ADMIN]
Do you grant the above permissions? [y/N] y
0.14-dev: Pulling from msterin/docker-volume-vsphere
7caaf7a208e6: Download complete 
Digest: sha256:bdfc033a2080f3a59064b42b3bfd44ba0e70f36763033d9792836a9c33495fc1
Status: Downloaded newer image for msterin/docker-volume-vsphere:0.14-dev
Installed plugin msterin/docker-volume-vsphere:0.14-dev
root@ubuntu-test:~# docker volume ls 
DRIVER              VOLUME NAME
vsphere:latest      vol1@datastore7
root@ubuntu-test:~# docker volume create -d vsphere vol2
vol2
root@ubuntu-test:~# docker volume ls 
DRIVER              VOLUME NAME
vsphere:latest      vol1@datastore7
vsphere:latest      vol2@datastore7
`

Comment (shuklanirdesh82) [inline vmdk_plugin/utils/test_util/VolumeCreationTestUtil.go]: @msterin 

```
root@sc-rdops-vm18-dhcp-57-89:~# docker plugin ls
ID                  NAME                DESCRIPTION                           ENABLED
beadd1d1d1b5        vsphere:latest      VMWare vSphere Docker Volume plugin   true
root@sc-rdops-vm18-dhcp-57-89:~# docker volume create -d vsphere vol2
Error response from daemon: create vol2: Post http://%2Frun%2Fdocker%2Fplugins%2Fvsphere.sock/VolumeDriver.Create: dial unix /run/docker/plugins/vsphere.sock: connect: no such file or directory
root@sc-rdops-vm18-dhcp-57-89:~# docker volume create -d vsphere:latest vol3
vol3

```

File: .drone.yml [hunk 4]
```diff
@@ -100,32 +103,37 @@ build:
...
+      - echo "=> Build and push vDVS managed plugin" `date` > /dev/null
       - #
       - export GOVC_URL=$GOVC_URL_6_5
       - export VM1=`$GOVC_GET_IP photon.vmfs`
       - export VM2=`$GOVC_GET_IP Ubuntu.16.10`
-      - ./misc/drone-scripts/deploy-and-test-wrapper.sh $GOVC_URL_6_5 $VM1 $VM2 $$BUILD_NUMBER < /dev/null
+      - #
+      - ./misc/drone-scripts/deploy-and-test-wrapper.sh deployplugin $GOVC_URL_6_5 $VM1 $VM2 $$BUILD_NUMBER < /dev/null
+      - #
+      - echo "=> Running tests on ESX_6.5" `date` > /dev/null
```
Comment (msterin) [inline .drone.yml]: This is fine, but do we also print  the actual (vmware -l) versoin in the ESX log.?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -169,10 +169,14 @@ Environment variables:
 - You **need** to set ESX and either VM_IP (in which case we'll use 1 VM) or
 both VM1 and VM2 environment variables
 
+- You need to set `DOCKER_HUB_REPO` as environment variable to complete our move to use [managed plugin](https://github.com/vmware/docker-volume-vsphere/blob/master/plugin/Makefile#L30).
```
Comment (msterin) [inline CONTRIBUTING.md]: Please add ***The build will use your user name (the output of 'whoami`) to decide on the DockerHub repo name. 
If you want to use another DockerHub repo*** you need to set ...

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -169,10 +169,14 @@ Environment variables:
...
+
+**Note**: You need to manually remove older rpm/deb installation from your test VMs. With [PR 1163](https://github.com/vmware/docker-volume-vsphere/pull/1163), our build/deployment script start using managed plugin approach.
```
Comment (msterin) [inline CONTRIBUTING.md]: I assume this is in "upgrade" section  ? Then it's NOT note. It's a sequence 
(1) remove using dpkg/yum (Example:) (2) install using docker

File: .drone.yml (hunk 4 above)
Comment (shuklanirdesh82) [inline .drone.yml]: Yes we are. https://github.com/vmware/docker-volume-vsphere/blob/master/misc/drone-scripts/deploy-and-test-wrapper.sh#L54

Comment (shuklanirdesh82) [inline .drone.yml]: Yes we are. https://github.com/vmware/docker-volume-vsphere/blob/master/misc/drone-scripts/deploy-and-test-wrapper.sh#L54

File: CONTRIBUTING.md (hunk 5 above)
Comment (shuklanirdesh82) [inline CONTRIBUTING.md]: done!

File: CONTRIBUTING.md (hunk 6 above)
Comment (shuklanirdesh82) [inline CONTRIBUTING.md]: I am going to generate a new PR to talk about upgrade scenario, this is just a note for code contributor for our repo.

I will add the sequence you have mentioned at the FAQ page under upgrade section.

**Pull request merged**